pkg/reconciler/instances/eventing/preaction/removenatsoperatorstep.go

```diff
@@ -90,10 +93,47 @@ func (r *removeNatsOperatorStep) removeNatsOperatorResources(context *service.Ac
 	// set the right eventing name, which went lost after rendering
 	manifest.Manifest = strings.ReplaceAll(manifest.Manifest, natsSubChartPath, eventingNats)
 
+	clientSet, err := kubeClient.Clientset()
+	tracker, err := progress.NewProgressTracker(clientSet, logger, progress.Config{Interval: progressTrackerInterval, Timeout: progressTrackerTimeout})
+	if err != nil {
+		return err
+	}
+
 	logger.Info("Removing nats-operator chart resources")
 	// remove all the existing nats-operator resources, installed via charts
-	_, err = kubeClient.Delete(context.Context, manifest.Manifest, namespace)
-	return err
+	statefulSet, err := getStatefulSet(context, kubeClient, eventingNats)
+	if err != nil {
+		return err
+	}
+
+	m := []byte(manifest.Manifest)
+	unstructs, err := kubernetes.ToUnstructured(m, true)
```

Comment:
(nit) Rename the `unstructs` to something more meaningful like `natsK8sResources` or `natsOperatorResources` etc.

Reply:
renamed